plugins/chain_plugin/tracked_votes.cpp

```diff
@@ -0,0 +1,82 @@
+#include <eosio/chain_plugin/tracked_votes.hpp>
+
+#include <shared_mutex>
+
+using namespace eosio;
+using namespace eosio::chain::literals;
+
+namespace eosio::chain_apis {
+   /**
+    * Implementation details of the last tracked cache
+    */
+   struct tracked_votes_impl {
+      tracked_votes_impl(const chain::controller& controller)
+      :controller(controller)
+      {}
+
+      // Cache to store last vote information for each known finalizer.
+      // A map of finalizer public key --> vote info.
+      std::unordered_map<std::string, tracked_votes::vote_info> last_votes;
```

Comment:
This is not thread safe. This can be called from multiple read-only threads at the same time. Need to add a mutex around the access/update of `last_votes`.

Reply:
`last_votes` is only updated by `on_accepted_block` which runs on the main thread, which does not interfere with read-only threads. That's why I did not use a mutex.